plugin_runner/plugin_runner.py

```diff
@@ -182,6 +189,38 @@ async def ReloadPlugins(
             yield ReloadPluginsResponse(success=True)
 
 
+def validate_effects(effects: list[Effect]) -> list[Effect]:
+    """Validates the effects based on predefined rules."""
+    autocomplete_effects = [
+        effect for effect in effects if effect.type == EffectType.AUTOCOMPLETE_SEARCH_RESULTS
+    ]
+
+    if len(autocomplete_effects) > 1:
+        log.warning("Only one AUTOCOMPLETE_SEARCH_RESULTS effect is allowed.")
+        return []
+
+    return effects
+
+
+def apply_effects_to_context(effects: list[Effect], event: Event) -> Event:
+    """Applies AUTOCOMPLETE_SEARCH_RESULTS effects to the event context.
+    If we are dealing with a search event, we need to update the context with the search results.
+    """
+    event_name = event.name
+
+    # Skip if the event is not a search event
+    if not event_name.endswith("__PRE_SEARCH") and not event_name.endswith("__POST_SEARCH"):
```

Comment:
Not a big deal right now, but I envision future non-search scenarios where one might want to mutate the context.

Reply:
Most likely yes, but we can adjust when the time comes.